Makefile

```diff
@@ -23,12 +35,12 @@ GIT_VERSION?=$(shell git describe --tags --dirty)
 docker-image: image.created
 image.created: dist/kube-policy-controller
 	# Build the docker image for the policy controller.
-	docker build -t $(CONTAINER_NAME) .
+	docker build -t $(CONTAINER_NAME) -f Dockerfile$(ARCHTAG) .
 	touch image.created
 
 dist/kube-policy-controller:
-	$(MAKE) OS=linux ARCH=amd64 binary-containerized
```

Comment:
I think I'd prefer an approach with a separate target, or just qualifying each target with `$ARCHTAG`.

For example, what if we had targets:

`dist/kube-policy-controller-linux-ppc64le`  - builds the binary, calling through to `binary-containerized` with the right args.
`image.created$ARCHTAG` - builds the docker image, depends on `dist/kube-policy-controller-linux-ppc64le`

And then the `docker-image` target could depend on `image.created$ARCHTAG`

Then the Dockerfile would just `ADD dist/kube-policy-controller-linux-ppc64le`.  I think this is more straighforward and less error prone (e.g. no chance of picking up the wrong stale binary from a previous build as part of the image build). WDYT? Worth it?

Reply:
Thank you for the review Casey.  I like the idea of using image.created$(ARCHTAG).  
Is your concern of picking up a stale binary for the case where a single environment is building multiple archs? Or am I misunderstanding?

How about this:

docker-image: image.created$(ARCHTAG)

image.created$(ARCHTAG): dist/kube-policy-controller$(ARCHTAG)
        # Build the docker image for the policy controller.
        docker build -t $(CONTAINER_NAME) -f Dockerfile$(ARCHTAG) .
        touch $@

dist/kube-policy-controller$(ARCHTAG):
        $(MAKE) OS=linux ARCH=$(ARCH) binary-containerized
<....>
clean:
        rm -rf dist image.created$(ARCHTAG)

Both Dockerfiles would need a change to pickup kube-policy-controller$(ARCHTAG) and link or rename it to kube-policy-controller.
WDYT?

Reply:
> Is your concern of picking up a stale binary for the case where a single environment is building multiple archs?

Yeah exactly. I'm also imagining a future where the standard release process produces binaries for multiple architectures.

What you posted looks like what I was expecting :)